src/mon/MonClient.cc

```diff
@@ -436,6 +452,15 @@ void MonClient::handle_monmap(MMonMap *m)
 void MonClient::handle_config(MConfig *m)
 {
   ldout(cct,10) << __func__ << " " << *m << dendl;
+
+  if (want_bootstrap_config) {
+    // get_monmap_and_config is waiting for config which it will apply
+    // synchronously
+    bootstrap_config = ceph::ref_t<MConfig>(m, false);
```

Comment:
Again, just worrying a little how safe it is to do without holding `monc_lock`?

Reply:
This function is called from `ms_dispatch` w/ `monc_lock` held

Reply:
Ah, thanks. I get used to see "is locked" assert in such cases, which makes things much clear.